mlir/lib/Dialect/Rock/Transforms/ConvToGemm.cpp

```diff
@@ -439,7 +439,10 @@ LogicalResult makeToLayoutLikeFromLayoutAlong(
   bool inOrder = true;
   size_t prevIndex = 0;
   for (StringAttr expected : expectedOrder) {
-    size_t thisIndex = toLayoutIdxs.find(expected)->getSecond();
+    auto foundp = toLayoutIdxs.find(expected);
+    assert(foundp != toLayoutIdxs.end());
```

Comment:
I had situations that wouldn't find the expected element in the toLayoutIdxs, causing the ->getSecond() to barf obscurely.

Reply:
Put a string on the assert?

Reply:
... Heck, this can be triggered by user input, it should be a `return failure()` instead